reo/utilities.py

```diff
@@ -253,35 +253,103 @@ def check_common_outputs(Test, d_calculated, d_expected):
-    :param relative_periods_df: pandas DataFrame with columns=["Month", "Start Week of Month", "Start Day of Week", "Start Hour", "Duration"] of length N periods.
-        All of the value types are integers **except** "Duration" which is of type timedelta (string equivalent to timedelta format)
-    :return: hourly_list: 8760 profile with 1.0 for timesteps defined in relative_periods_df, else 0.0.
-    NOTE: Blank
-    '''
-    if calendar.isleap(year):  # Remove last day of the year if leap year
+    :param year: year for applying consecutive_periods changes based on year and leap years (cut off 12/31/year)
+    :param consecutive_periods: either list of dictionaries where each dict defines a period (keys = "month", "start_week_of_month", "start_day_of_week", "start_hour", "duration_hours"; length N periods)
+        OR can be a Pandas DataFrame with columns equivalent to the dict keys in which case it gets converted to list_of_dict. All of the value types are integers.
+    :return: year_profile_hourly_list: 8760 profile with 1.0 for timesteps defined in consecutive_periods, else 0.0.
+    :return: errors_list used in validators.py - errors related to the input consecutive_periods and the year's calendar
+    """
+    errors_list = []
+    # Create datetime series of the year, remove last day of the year if leap year
+    if calendar.isleap(year):
         end_date = "12/31/"+str(year)
     else:
         end_date = "1/1/"+str(year+1)
     dt_profile = pd.date_range(start='1/1/'+str(year), end=end_date, freq="1H", closed="left")
     year_profile_hourly_series = pd.Series(np.zeros(8760), index=dt_profile)
-    for i in range(len(relative_periods_df)):
-        start_month = relative_periods_df["Month"][i]
-        start_week = relative_periods_df["Start Week of Month"][i]
-        start_weekday = relative_periods_df["Start Day of Week"][i]  # Monday is 0, Sunday is 6
-        day_of_month = calendar.Calendar().monthdayscalendar(year=year,month=start_month)[start_week][start_weekday]
-        hour_of_day = relative_periods_df["Start Hour"][i]
-        start_datetime = datetime.datetime(year=year, month=start_month, day=day_of_month, hour=hour_of_day)
-        duration = pd.to_timedelta(relative_periods_df["Duration"][i])
-        year_profile_hourly_series[start_datetime:start_datetime+duration-datetime.timedelta(hours=1)] = 1.0
+    
+    # Check if the consecutive_periods is a list_of_dict or other (must be Pandas DataFrame), and if other, convert to list_of_dict
+    if not isinstance(consecutive_periods, list):
+        consecutive_periods = convert_dataframe_to_list_of_dict(consecutive_periods)
+
+    day_of_week_name = ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"]
+    for i in range(len(consecutive_periods)):
+        start_month = int(consecutive_periods[i]["month"])  # One-indexed both user input and Calendar package
```

Comment:
you are expecting certain keys here (i.e. month, start_hour), I would suggest using .get in case the user did not supply them. You might also add to validators.py to check that incoming data is formatted as expected

Reply:
You may have seen after this comment that I'm only ever calling this function after validating the inputs. If the inputs are not validated prior to the function, e.g. if one of the keys is missing, then this function will error without a useful exception.

Reply:
Did you see the comment in validators where it didn't seem to check that all required keys were provided though? I think it only validated the keys that were given

Reply:
You mean 2 below this comment? I added validation for not all keys required now.

Reply:
Looks good, sorry github isn't showing the updates you made clearly